ReSharper.FSharp/src/FSharp.Psi/src/Util/FSharpExpressionUtil.cs

```diff
@@ -35,5 +36,58 @@ public static class FSharpExpressionUtil
 
     public static readonly Func<IFSharpExpression, bool> IsSimpleValueExpressionFunc = IsSimpleValueExpression;
     public static readonly Func<IFSharpExpression, bool> IsLiteralExpressionFunc = IsLiteralExpression;
+
+    // TODO: change name
+    public static IFSharpExpression GetOuterMostParentExpression(this IFSharpExpression expression)
+    {
+      var currentExpr = expression;
+      while (true)
+      {
+        if (MatchClauseListOwnerExprNavigator.GetByClauseExpression(currentExpr) is { } matchExpr)
+        {
+          currentExpr = matchExpr as IFSharpExpression;
+          continue;
+        }
+        
+        if (SequentialExprNavigator.GetByExpression(currentExpr) is { } seqExpr &&
+            seqExpr.ExpressionsEnumerable.Last() == currentExpr)
+        {
+          currentExpr = seqExpr;
+          continue;
+        }
+
+        if ((IfThenElseExprNavigator.GetByBranchExpression(currentExpr)) is {ElseExpr: { }} ifExpr)
+        {
+          currentExpr = ifExpr;
+          continue;
+        }
+
+        if (LetOrUseExprNavigator.GetByInExpression(currentExpr) is { } letOrUseExpr)
+        {
+          currentExpr = letOrUseExpr;
+          continue;
+        }
+
+        if (LambdaExprNavigator.GetByExpression(currentExpr) is { } lambdaExpr)
+        {
+          currentExpr = lambdaExpr;
+          continue;
+        }
+
+        if (TupleExprNavigator.GetByExpression(currentExpr) is { } tupleExpr)
+        {
+          currentExpr = tupleExpr;
+          continue;
+        }
```

Comment:
Could you show an example where we need to go up from a tuple expression?

Reply:
See test `Tuple return type.fs`

Reply:
Another interesting case:
```fsharp
let x: int -> int = 1, 1
```

Reply:
We shouldn't go up here, but it's OK to go up in the quick fix (as you already seem to do).